test/correctness/fuzz_simplify.cpp

```diff
@@ -36,6 +37,21 @@ Type random_type(int width) {
     return T;
 }
 
+int get_random_divisor(Type t) {
+    // extract_lane won't work on the ramp of boolean type with width > 2.
```

Comment:
This seems like a bug if I'm interpreting this comment correctly. Should it have an issue?

Reply:
I think the problem here is expression like ramp((uint1)a, (uint1)b, 4), for which one of the components will be (uint1)a + 2 * (uint1). This multiplication by 2 seems to be problematic and triggers `Integer constant 2 will be implicitly coerced to type uint1, which changes its value to (uint1)0.`error. I assumed that this is expected behaviour, but if not I would definitely file a bug.

Reply:
Oh, this makes sense, just the comment is a bit confusing.

It does seem like there's a bit of an issue here. It seems like either ramps of uint1 should be OK (because uint overflow is well defined), and then `extract_lane` should work. Or ramps of uint1 are not OK, and in that case, `extract_lane` is not broken (the comment is confusing).

But I'm not sure it's something we should spend time on (seems like a case we don't ever actually hit)

Reply:
Maybe the easiest thing to do is actually just try fixing `extract_lane` so it doesn't generate this error (i.e. explicitly implement overflow for unsigned types). If that works, then there is no special case/confusing comment required.

Reply:
Hm, it was enough to put explicit cast. Also, removed special handling of uint1 from fuzz_simplify.